reproducible/ubuntu/mknodeimage.sh

```diff
@@ -1,19 +1,29 @@
 #!/bin/bash
 set -ex
 
+VERSION=$1
+
 ln -s /usr/bin/python2.7 /usr/bin/python
 
 WORKDIR="/workspace/nodejs_build"
 OUTPUTDIR="$WORKDIR/nodejs"
 mkdir -p "$OUTPUTDIR"
 cd "$WORKDIR"
 
-tar -xvf /node-v6.11.4.tar.gz
-cd node-v6.11.4
-./configure --prefix="$OUTPUTDIR"
-make -j4
-make install
+NPROCS="$(grep -c ^processor /proc/cpuinfo)"
+MAX_PROCS="8"
+
+mkdir -p /nodejs
+# TODO(aprindle) add SHA verification
+curl -O "https://nodejs.org/dist/$VERSION/node-$VERSION.tar.gz"
+
+tar -xvf "node-$VERSION.tar.gz"
+cd "node-$VERSION"
+./configure --prefix="$OUTPUTDIR" &> /dev/null
+# Sets number of cores to use for make, maximum of 4
+make -j$(($NPROCS>$MAX_PROCS?$MAX_PROCS:$NPROCS)) &> /dev/null
+make install &> /dev/null
 
 # pass -n to gzip to strip timestamps
 # strip the '.' with --transform thatp tar includes at the root to build nodejs
-GZIP="-n" tar --numeric-owner -czf /workspace/nodejs.tar.gz -C "$OUTPUTDIR" . --transform='s,^./,,' --mtime='1970-01-01'
+GZIP="-n" tar --numeric-owner -czf /workspace/node_${VERSION}_compiled.tar.gz -C "$OUTPUTDIR" . --transform='s,^./,,' --mtime='1970-01-01'
```

Comment:
not sure you need the transform= part, I think that's only important for rootfs tarballs.

Reply:
The node installation is a rootfs tarball created from 'make install' into a separated dir.